In [4]:
# First, install the lightgbm package
!pip install lightgbm

# Then run your original code
import pandas as pd
import glob
import lightgbm as lgb
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score, roc_auc_score

# 1. load 5 csv datasets
files = glob.glob('match_data_*.csv')
df = pd.concat([pd.read_csv(f) for f in files], ignore_index=True)

print(f"row number: {df.shape[0]}, feature amount: {df.shape[1]}")

# 2. delete ID columns
if 'matchId' in df.columns:
    df = df.drop(columns=['matchId'])

# 3. separate varaibles
y = df['win']
X = df.drop(columns=['win'])

categorical_cols = X.select_dtypes(include=['object']).columns.tolist()
for col in categorical_cols:
    X[col] = X[col].astype('category')

# 4. 80% training, 20% test
X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42, stratify=y
)

# 5. LightGBM
model = lgb.LGBMClassifier(random_state=42, n_jobs=-1)
model.fit(X_train, y_train)

# 6. results
y_pred = model.predict(X_test)
y_pred_proba = model.predict_proba(X_test)[:, 1]

acc = accuracy_score(y_test, y_pred)
auc = roc_auc_score(y_test, y_pred_proba)

print(f"baseline Accuracy: {acc:.4f}")
print(f"baseline AUC: {auc:.4f}")

row number: 19464, feature amount: 158
[LightGBM] [Info] Number of positive: 7870, number of negative: 7701
[LightGBM] [Info] Auto-choosing col-wise multi-threading, the overhead of testing was 0.003194 seconds.
You can set `force_col_wise=true` to remove the overhead.
[LightGBM] [Info] Total Bins 21790
[LightGBM] [Info] Number of data points in the train set: 15571, number of used features: 156
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.505427 -> initscore=0.021708
[LightGBM] [Info] Start training from score 0.021708
baseline Accuracy: 0.7120
baseline AUC: 0.7861


## results:
!!!
baseline Accuracy: 0.7120
baseline AUC: 0.7861

row number: 19464, feature amount: 158
[LightGBM] [Info] Number of positive: 7870, number of negative: 7701
[LightGBM] [Info] Total Bins 21790
[LightGBM] [Info] Number of data points in the train set: 15571, number of used features: 156
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.505427 -> initscore=0.021708
[LightGBM] [Info] Start training from score 0.021708
